webapp/src/app.tsx

```diff
@@ -1168,6 +1168,42 @@ export class ProjectView extends data.Component<IAppProps, IAppState> {
             })
     }
 
+    launchFullEditor() {
+        let sandbox = /sandbox=1/i.test(window.location.href)
+        let sandboxPub = /#sandbox:/i.test(window.location.href)
+        let sandboxProject = /#sandboxproject:/i.test(window.location.href)
+        Util.assert(sandbox || sandboxPub || sandboxProject);
+
+        let rootUrl = pxt.appTarget.appTheme.embedUrl;
+        if (!/\/$/.test(rootUrl)) rootUrl += '/';
+
+        const launchEditor = (url: string) => {
+            pxt.tickEvent("sandbox.launchexternaleditor");
+            let editUrl = `${rootUrl}${url}`;
+            window.open(editUrl, '_blank')
+        }
+
+        if (sandboxPub) {
+            let mpkg = pkg.mainPkg
+            let epkg = pkg.getEditorPkg(mpkg)
+            if (epkg.header.pubCurrent) {
+                launchEditor(`#pub:${epkg.header.pubId}`);
+            } else {
+                this.publishAsync()
```

Comment:
IMO, we should not publish to swap out of the sandbox. Always encode as project and pass along